src/anki/ankiclient.cpp

```diff
@@ -1384,6 +1386,19 @@ void AnkiClient::buildCommonNote(
         value.replace(REPLACE_CLOZE_PREFIX, clozePrefix);
         value.replace(REPLACE_CLOZE_SUFFIX, clozeSuffix);
         value.replace(REPLACE_FREQUENCIES,  frequencies);
+
+        value.replace(REPLACE_FREQ_HARMONIC_RANK, 
+                positiveIntToQString(frequencyHarmonic, 9999999));
+
+        value.replace(REPLACE_FREQ_HARMONIC_OCCU, 
+                positiveIntToQString(frequencyHarmonic, 0));
+
+        value.replace(REPLACE_FREQ_AVERAGE_RANK,  
+                positiveIntToQString(frequencyAverage,  9999999));
+
+        value.replace(REPLACE_FREQ_AVERAGE_OCCU,  
+                positiveIntToQString(frequencyAverage,  0));
+
```

Comment:
I was a bit conflicted on whether to repeat `frequencyXXX < 0 ? ["999999"|"0"] : QString::number(frequencyXXX);` four times or to consolidate that into a `positiveIntToQString` method.

Reply:
I think you made the right choice